src/SDKs/Azure.Base/data-plane/Azure.Base/ClientPipelineBuilderExtensions.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using Azure.Base.Http;
+using Azure.Base.Http.Pipeline;
+
+namespace Azure
+{
+    public static class ClientPipelineBuilderExtensions
+    {
+        public static HttpPipeline Build(this HttpPipelineOptions options, string componentName, string componentVersion)
+        {
+            if (!options.DisableTelemetry)
+            {
+                options = options.Clone();
+                options.AddHeaders(HttpHeader.Common.CreateUserAgent(componentName, componentVersion, options.ApplicationId));
+            }
+
+            var transport = options.TransportPolicy();
+
+            HttpPipelinePolicy currentPolicy = transport;
+
+            foreach (var section in options.PipelinePolicies.Reverse().GroupBy(registration => registration.Section))
+            {
+                foreach (var pipelineBuilderPipelinePolicy in section)
+                {
+                    currentPolicy = pipelineBuilderPipelinePolicy.PolicyFactory(currentPolicy);
+                }
+            }
+
+            return new HttpPipeline(transport, currentPolicy);
+        }
+
+        public static HttpPipelineOptions AddLogging(this HttpPipelineOptions options, int[] excludeCodes = null)
+        {
+            return options.Append(HttpPipelineSection.PostRetry, next => new LoggingPolicy(next, excludeCodes));
+        }
+
+        public static HttpPipelineOptions AddFixedRetry(this HttpPipelineOptions options, int maxRetries, TimeSpan delay, params int[] retriableCodes)
```

Comment:
If people want to change max retries only, how will they know what status codes they should pass?

Reply:
How do they do it in the current design? I'm not trying to fix every problem we have here.